SYCL/SubGroup/info.cpp

```diff
@@ -60,11 +60,18 @@ int main() {
     if (std::find(Vec.begin(), Vec.end(), "cl_intel_required_subgroup_size") !=
         std::end(Vec)) {
       auto sg_sizes = Device.get_info<info::device::sub_group_sizes>();
+
+      Res = Kernel.get_info<info::kernel_device_specific::max_sub_group_size>(
+          Device);
+      bool Expected =
+          std::find(sg_sizes.begin(), sg_sizes.end(), Res) != sg_sizes.end();
```

Comment:
I think we should ensure that it's actually the maximum element of the range.

```suggestion
          bool Expected = (Res == *std::max_element(sg_sizes.begin(), sg_sizes.end());
```

Reply:
I disagree.

`info::device::sub_group_sizes` is the list of all sub-group sizes supported by the device, whereas `info::kernel_device_specific::max_sub_group_size` is the maximum sub-group size supported by a specific kernel. The kernel-specific number may be lower than the device's maximum sub-group size (e.g. if the kernel uses a feature that only works with certain sub-group sizes), so I think the current test is correct.

Reply:
Yes, I've realized that after reading the PR in intel/llvm but forgot to return to here. I think we need to add a comment here explaining just that.

Reply:
A comment has been added.